src/UI/DataAutofill.js

```diff
@@ -59,18 +54,12 @@ ${escapeXML(singleData[config.displaySubtitlePropName] || config.displaySubtitle
 
         this.autofillButtons.forEach((btn) => {
             this.registerClickableButton(btn, () => {
-                this.interface[`${config.autofillMethod}`](btn.id).then(({success}) => {
-                    if (success) {
-                        this.fillForm(success)
-                    }
-                })
+                callbacks.onSelect(btn.id)
             })
         })
 
         this.init()
-    }
-    async fillForm (data) {
-        this.interface.selectedDetail(data, this.config.type)
+        return this
```

Comment:
I'm interested why returning `this` is a pattern? Constructors are happy without right?

Reply:
I think it's used so you can call a method right on instantiation `new Something().someMethod()`.

Reply:
@jonathanKingston this is no longer a constructor, but yeah the pattern is to allow chaining if needed